### ISSUE TITLE
Suppose Python 3 type annotations?

### ISSUE THREAD

**Issue by thesamovar** [brian-team/brian2#598]
As an alternative to declaring types with a decorator or Function.

**Comment by mstimberg** [comment]
I agree that could be something nice to support (but then: for types or for units?) -- but let's postpone it to post 2.0 :)

**Comment by mstimberg** [comment]
To make this clearer, let's keep this issue for the type annotation mechanism for physical units in user-defined functions. I'll open another one for using type annotations in the "classical" way.
